cli/cli/commands/cluster/set/set.go

```diff
@@ -60,50 +59,25 @@ func run(ctx context.Context, flags *flags.ParsedFlags, args *args.ParsedArgs) e
 		return nil
 	}
 
-	engineManagerOldCluster, err := engine_manager.NewEngineManager(ctx)
-	if err != nil {
-		return stacktrace.Propagate(err, "An error occurred creating an engine manager.")
-	}
-	if err = engineManagerOldCluster.StopEngineIdempotently(ctx); err != nil {
-		logrus.Warnf("Kurtosis was not able to stop a potentially running engine on the cluster '%s'. This is not "+
-			"critical, it might be that the current cluster is not reachable anymore. In the worst case scenario, the "+
-			"engine will continue running in cluster '%s' until Kurtosis is switched back to it.", clusterPriorToUpdate,
-			clusterPriorToUpdate)
-	}
-
 	if err = clusterSettingStore.SetClusterSetting(clusterName); err != nil {
 		return stacktrace.Propagate(err, "Failed to set cluster name to '%v'.", clusterName)
 	}
 	defer func() {
 		if clusterUpdateSuccessful {
 			return
 		}
+		if clusterPriorToUpdate == noClusterSetting {
```

Comment:
I think that this defer doesn't make any sense anymore, we could leave it if we can prevent someone else forget to add it if they add new lines after `clusterSettingStore.SetClusterSetting` but we could also remove it for making the code simpler, I leave it to you.